std/algorithm.d

```diff
@@ -1391,6 +1394,145 @@ unittest //13304
     static assert(is(typeof(reduce!((a, b)=>a+b)(data))));
 }
 
+
+// each
+/**
+Eagerly iterates over $(D r) and calls $(D fun) over _each element.
+
+Example:
+---
+void deleteOldBackups()
+{
+    import std.algorithm, std.datetime, std.file;
+    auto cutoff = Clock.currTime() - 7.days;
+    dirEntries("", "*~", SpanMode.depth)
+        .filter!(de => de.timeLastModified < cutoff)
+        .each!remove();
+}
+---
+
+If the range supports it, the value can be mutated in place. Examples:
+---
+arr.each!((ref a) => a++);
+arr.each!"a++";
+---
+
+If no predicate is specified, $(D each) will default to doing nothing
+but consuming the entire range. $(D .front) will be evaluated, but this
+can be avoided by explicitly specifying a predicate lambda with a
+$(D lazy) parameter.
+
+$(D each) also supports $(D opApply)-based iterators, so it will work
+with e.g. $(XREF parallelism, parallel).
+
+See_Also: $(XREF range,tee)
+
+ */
+template each(alias fun = "a")
```

Comment:
A predicate is a subset of functions that return either true or false. I think his choice is correct here, but it should be changed everywhere `fun` is called a predicate.